test/linalg/special.jl

```diff
@@ -9,26 +9,22 @@ srand(1)
 debug && println("Test interconversion between special matrix types")
 let a=[1.0:n;]
    A=Diagonal(a)
-   for newtype in [Diagonal, Bidiagonal, SymTridiagonal, Tridiagonal, LowerTriangular, UpperTriangular, Matrix]
+   for newtype in [Diagonal, Bidiagonal, SymTridiagonal, Tridiagonal, Matrix]
        debug && println("newtype is $(newtype)")
        @test full(convert(newtype, A)) == full(A)
    end
-   for newtype in [Base.LinAlg.UnitUpperTriangular, Base.LinAlg.UnitLowerTriangular]
-       @test_throws ArgumentError convert(newtype, A)
-       @test full(convert(newtype, Diagonal(ones(n)))) == eye(n)
-   end
 
    for isupper in (true, false)
        debug && println("isupper is $(isupper)")
        A=Bidiagonal(a, [1.0:n-1;], isupper)
-       for newtype in [Bidiagonal, Tridiagonal, isupper ? UpperTriangular : LowerTriangular, Matrix]
+       for newtype in [Bidiagonal, Tridiagonal, Matrix]
            debug && println("newtype is $(newtype)")
            @test full(convert(newtype, A)) == full(A)
            @test full(newtype(A)) == full(A)
        end
        @test_throws ArgumentError convert(SymTridiagonal, A)
        A=Bidiagonal(a, zeros(n-1), isupper) #morally Diagonal
-       for newtype in [Diagonal, Bidiagonal, SymTridiagonal, Tridiagonal, isupper ? UpperTriangular : LowerTriangular, Matrix]
+       for newtype in [Diagonal, Bidiagonal, SymTridiagonal, Tridiagonal, Matrix]
            debug && println("newtype is $(newtype)")
            @test full(convert(newtype, A)) == full(A)
            @test full(newtype(A)) == full(A)
```

Comment:
Please see the response above. `Diagonal(A)`/`convert(Diagonal, A)`, `SymTridiagonal(A)`/`convert(SymTridiagonal, A)`, and `Tridiagonal(A)`/`convert(Tridiagonal, A)` appear to test separate code paths as well (which seems a bit odd, the relevant methods would probably benefit from a bit of refactoring). Thoughts? Thanks!